[Doc] GraphStorm API example notebook 6 converting API examples to run with GraphStorm CLIs (#1049)
awslabs/graphstorm: docs/source/api/notebooks/index.rst

These examples utilize GraphStorm APIs, such as ``graphstorm``, ``graphstorm.dataloading.GSgnnDataset``, ``graphstorm.dataloading.GSgnnNodeDataLoader``, ``graphstorm.trainer.GSgnnNodePredictionTrainer``, and ``graphstorm.inference.GSgnnNodePredictionInferrer``, to form the training and infernece pipeline. In terms of the GNN models, users can refer to the `demo_model.py <https://github.com/awslabs/graphstorm/blob/main/docs/source/api/notebooks/demo_models.py>`_ file in which all models are created by using GraphStorm APIs.

> These notebooks can run in the GraphStrom Standalone mode, i.e., on a single CPU or GPU of a single machine, which is good for quick model development and debugging on a relatively small graph dataset. To fully leverage GraphStorm's distributed model training and inference capacity, users can follow the guidelines shown in `Notebook 6: Converting Customized Model Notebooks to Using GraphStorm CLIs <https://github.com/awslabs/graphstorm/blob/main/docs/source/api/notebooks/Notebook_6_API2CLI.ipynb>`_ and choose proper distributed environment to run the customized models for their enterprise-level graph datasets. Users can refer to the `demo_run_train.py <https://github.com/awslabs/graphstorm/blob/main/docs/source/api/notebooks/demo_run_train.py>`_ and `demo_run_infer.py <https://github.com/awslabs/graphstorm/blob/main/docs/source/api/notebooks/demo_run_infer.py>`_ for examples of the converted model Python files to be used by GraphStorm CLIs.

Reviewer: ```suggestion
These notebooks can run with the GraphStrom Standalone mode, i.e., on a single CPU or GPU of a single machine. To fully leverage GraphStorm's distributed model training and inference capability, users can follow the guidelines shown in `Notebook 6: Converting Customized Model Notebooks to Using GraphStorm CLIs <https://github.com/awslabs/graphstorm/blob/main/docs/source/api/notebooks/Notebook_6_API2CLI.ipynb>`_ and choose a proper distributed environment to run the custom models for their enterprise-level graph datasets. Users can refer to the `demo_run_train.py <https://github.com/awslabs/graphstorm/blob/main/docs/source/api/notebooks/demo_run_train.py>`_ and `demo_run_infer.py <https://github.com/awslabs/graphstorm/blob/main/docs/source/api/notebooks/demo_run_infer.py>`_ as examples of the converted model Python files to be used by GraphStorm CLIs.
```

Reviewer: What does `the converted model Python files` mean?
